pkg/kyverno/common/common.go

```diff
@@ -520,7 +520,7 @@ func MutatePolices(policies []*v1.ClusterPolicy) ([]*v1.ClusterPolicy, error) {
 
 // ApplyPolicyOnResource - function to apply policy on resource
 func ApplyPolicyOnResource(policy *v1.ClusterPolicy, resource *unstructured.Unstructured,
-	mutateLogPath string, mutateLogPathIsDir bool, variables map[string]string, policyReport bool, namespaceSelectorMap map[string]map[string]string, stdin bool, rc *ResultCounts) (*response.EngineResponse, policyreport.Info, error) {
+	mutateLogPath string, mutateLogPathIsDir bool, variables map[string]string, policyReport bool, namespaceSelectorMap map[string]map[string]string, stdin bool, rc *ResultCounts, printPatchResource bool) ([]*response.EngineResponse, *response.EngineResponse, policyreport.Info, error) {
```

Comment:
Why this function is returning both`[]*response.EngineResponse` and `*response.EngineResponse` can it not be clubbed into `[]*response.EngineResponse` only ?

Reply:
> 
> 
> Why this function is returning both`[]*response.EngineResponse` and `*response.EngineResponse` can it not be clubbed into `[]*response.EngineResponse` only ?

Long before, at the starting when I cloned the repo, both responses were present and later on you removed it.  
And if we talk about clubbing then  

> 
> 
> > > Also, why the test cases are still failing? Is it dependent on any issue??
> > 
> > 
> > They are failing because those test cases do not contains `kind` in the result section of `test.yaml` file.
> 
> Is anyone working on it? Do you have a issue and PR raised for the same?

The  issue was raised earlier and @sonukushwaha403 took the charge for the issue. And he covered all policies  but forgot to cover  test cases.  @sonukushwaha403, can you complete the remaining part.

Reply:
can you fix the test cases?

Reply:
> > Why this function is returning both`[]*response.EngineResponse` and `*response.EngineResponse` can it not be clubbed into `[]*response.EngineResponse` only ?
> 
> Long before, at the starting when I cloned the repo, both responses were present and later on you removed it.
> And if we talk about clubbing then I have to check it whether it works or not.
> 
> > > > Also, why the test cases are still failing? Is it dependent on any issue??
> > > 
> > > 
> > > They are failing because those test cases do not contains `kind` in the result section of `test.yaml` file.
> > 
> > 
> > Is anyone working on it? Do you have a issue and PR raised for the same?
> 
> The [issue](https://github.com/kyverno/policies/pull/123) was raised earlier and @sonukushwaha403 took the charge for the issue. And he covered all policies but forgot to cover test cases. @sonukushwaha403, can you complete the remaining part.

why are you using `engineResponses`? It is appending mutate response and in that function we will get only one mutate response. Then why do we need to append it?

Reply:
I am trying to make one common response i.e. `engineResponses` so that it  can holds response of any type of responses i.e. mutate, validate or generate depending on the type of policy being used.